### PR TITLE
C24_WMDE_iPad_DE_01

### PR DESCRIPTION
**Pull request by Sperling-0**
https://phabricator.wikimedia.org/T376593

### CODE REVIEW

File: banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss [hunk 1]
```diff
@@ -5,12 +5,13 @@
 	$select-group-button: true,
```
Comment (Abban) [inline banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:5]: When https://github.com/wmde/fundraising-banners/pull/579 is merged the payment logos here will break. You should be able to already add `$select-group-image-label: true` to this list.

File: banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss [hunk 2]
```diff
@@ -5,12 +5,13 @@
...
 	$upgrade-to-yearly: true,
 	$fallback-banner: true,
+	$soft-close: true
 );
 @use '@src/components/BannerConductor/banner-transition';
 @use 'Banner';
 @use 'MainBanner' with (
 	$banner-height: 378px,
-	$form-width: 256px
+	$form-width: 35%
```
Comment (Abban) [inline banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:15]: Pixel width is used here is so the form remains consistent on all screen sizes. Is there a reason you changed it to a percentage?

File: src/themes/Treedip/ProgressBar/ProgressBar.scss [hunk 3]
```diff
@@ -42,7 +42,7 @@ $progress-bar-margin: 0 !default;
 		&-fill-wrapper {
 			position: relative;
 			background: var( --progress-bar-wrapper-background );
-			border: 3px var( --progress-bar-wrapper-border );
+			border: 3px var( --progress-bar-wrapper-border ) solid;
```
Comment (Abban) [inline src/themes/Treedip/ProgressBar/ProgressBar.scss]: This works, but the general convention for the order when defining border styles is `border: <size> <style> <colour>;`

```css
border: 3px solid var( --progress-bar-wrapper-border );
```

File: banners/pad/C24_WMDE_iPad_DE_01/messages_var.ts
Comment (Abban) [inline banners/pad/C24_WMDE_iPad_DE_01/messages_var.ts:1]: This is identical to `messages.js` so you can delete this and use the same file in both banners.

**Review by Abban — APPROVED** [review]
It's looking good, just needs a few tweaks.

**Comment by Sperling-0** [comment]
⚠️ Rebase this PR from latest main.

To get the latest UoF working:

- For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
- For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

**Comment by Sperling-0** [comment]
> ⚠️ Rebase this PR from latest main.
> 
> To get the latest UoF working:
> 
>     * For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
> 
>     * For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

✅ Done

File: banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss (hunk 2 above)
Comment (Sperling-0) [reply banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:15]: I did not change it. It came from `styles_var.scss` of `C24_WMDE_iPad_00`.

**Comment by moiikana** [comment]
consider rebasing because https://github.com/wmde/fundraising-banners/pull/597 got merged

**Comment by Sperling-0** [comment]
> consider rebasing because #597 got merged

✅ Done

File: banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import { WindowSizeIssueChecker } from '@src/utils/SizeIssueChecker/WindowSizeIssueChecker';
+import TranslationPlugin from '@src/TranslationPlugin';
+import { LegacyTrackerWPORG } from '@src/tracking/LegacyTrackerWPORG';
+import eventMappings from './event_map';
+import { Translator } from '@src/Translator';
+import DynamicTextPlugin from '@src/DynamicTextPlugin';
+import { LocalImpressionCount } from '@src/utils/LocalImpressionCount';
+
+// Channel specific form setup
+import { createFormItems } from './form_items_var';
```
Comment (moiikana) [inline banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts:21]: the ticket doesn't say anywhere that the variant should have different form items. is this a bug that was missed?

File: banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss (hunk 2 above)
Comment (moiikana) [reply banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:15]: but in ipad00 it says 256px instead of 35%.
The 35% got introduced in this PR

**Review by moiikana — CHANGES_REQUESTED** [review]

File: banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss (hunk 2 above)
Comment (moiikana) [reply banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:15]: maybe we can try going back to the pixel value here

File: banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts (hunk 4 above)
Comment (Sperling-0) [reply banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts:21]: Screenshot from 2024-10-29 14-25-12

Please see the last point for the VAR banner.

That is what was talked about in the daily today. For ipad-01 and wpde-mob-01 banners the ticket kept renewing and new changes kept coming.

**Comment by Sperling-0** [comment]
> [!Warning]
> Please do not merge this. The tests (timer and cookie image) are filing locally for me. So maybe this needs investigation.

File: banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts (hunk 4 above)
Comment (moiikana) [reply banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts:21]: thanks! didn't see that.

**Comment by Sperling-0** [comment]
> Warning
> 
> Please do not merge this. The tests (timer and cookie image) are filing locally for me. So maybe this needs investigation.

> [!NOTE]  
> After the rebase to the latest main branch. All the tests are passing!!!

**Review by gbirke — APPROVED** [review]

**Comment by gbirke** [comment]
Could you move `tests/banners/pad/components` to `tests/banners/pad/C24_WMDE_iPad_DE_01/components` to prepare for #594 ? You can do it in a separate commit

**Review by moiikana — APPROVED** [review]

**Comment by Sperling-0** [comment]
> Could you move `tests/banners/pad/components` to `tests/banners/pad/C24_WMDE_iPad_DE_01/components` to prepare for #594 ? You can do it in a separate commit

✅ Done

**Pull request merged**